platform/x11/os_x11.cpp

```diff
@@ -686,6 +686,16 @@ void OS_X11::get_fullscreen_mode_list(List<VideoMode> *p_list, int p_screen) con
 }
 
 void OS_X11::set_wm_fullscreen(bool p_enabled) {
+	if (p_enabled == true && is_window_resizable() == false) {
```

Comment:
You could save redundant operations with just `p_enabled && !is_window_resizable()` (same below).